src/Detector/Detector.php

```diff
@@ -25,6 +25,12 @@
 final class Detector{
 
 	private BitMatrix $matrix;
+	public int $topLeftX;
+	public int $topLeftY;
+	public int $topRightX;
+	public int $topRightY;
+	public int $bottomLeftX;
+	public int $bottomLeftY;
```

Comment:
We don't do public properties here :)

I think we can hand down the whole result from the finder pattern finder as the objects hold other useful values such as the estimated module size, so just add one property (with a getter) that holds the result array `private array|null $finderPatterns = null` (nullable in case it gets called from the getter before `Detector::detect()` was called).